connection.go

```diff
@@ -138,7 +139,10 @@ func (sc *snowflakeConn) exec(
 	}
 	logger.WithContext(ctx).Infof("Success: %v, Code: %v", data.Success, code)
 	if !data.Success {
-		return nil, (populateErrorFields(code, data)).exceptionTelemetry(sc)
+		errMutex.Lock()
+		err = (populateErrorFields(code, data)).exceptionTelemetry(sc)
+		errMutex.Unlock()
```

Comment:
Is it guaranteed even above line throw an error/exception it will come to this line? Basically in any situation, this mutex stay in locked stage without unlock() executed for any reason?

Reply:
I would rewrite this code as:
```
		errMutex.Lock()
		defer errMutex.Unlock()
		err = (populateErrorFields(code, data)).exceptionTelemetry(sc)
		return nil, err
```
What do you think?